src/main/host/syscall_handler.c

```diff
@@ -165,17 +165,26 @@ static void _syscallhandler_post_syscall(SysCallHandler* sys, long number,
         scr = syscallhandler_##s(sys, args);                                   \
         _syscallhandler_post_syscall(sys, args->number, #s, &scr);             \
         break
-#define HANDLE_NEW(s)                                                              \
-    case SYS_##s:                                                              \
-        _syscallhandler_pre_syscall(sys, args->number, #s);                    \
-        scr = newsyscallhandler_##s(sys, args);                                   \
-        _syscallhandler_post_syscall(sys, args->number, #s, &scr);             \
-        break
 #define NATIVE(s)                                                              \
     case SYS_##s:                                                              \
         debug("native syscall %ld " #s, args->number);                         \
         scr = (SysCallReturn){.state = SYSCALL_NATIVE};                        \
         break
+
+/* Comment out this line to use the C syscall handlers. */
+#define USE_RUST_SYSCALLS
```

Comment:
I'm not sure, do we want this as a command line option instead of #define?

Or maybe a better question is, what is the plan for eventually removing the c syscall handlers? Is it just until we feel the rust versions are "stable"?

Reply:
Yeah this may be better as the inverse (ex: `USE_C_SYSCALLS`) which defaults to unset, and then can be enabled as a gcc flag with `-D USE_C_SYSCALLS`.

Right now the Rust syscall handlers still depend on the C syscall handlers if the descriptor is a "legacy descriptor", so some of the C syscall handler code (for example `syscallhandler_read`) will have to stay until all of the descriptor types are implemented in Rust.

Reply:
It can now be set with a CLI flag. For example, `./setup build --debug --test --use-c-syscalls`.